spec/linter-view-spec.coffee

```diff
@@ -41,3 +46,55 @@ describe "LinterView:lint", ->
     waitsFor ->
       calls is linterClasses.length
     , "lint file finished"
+
+
+describe "LinterView", ->
+
+  lv = null
+  stub_editor = sinon.createStubInstance(Editor)
+  stub_editor.isAlive.returns(true)
+  messages = [
+    {line: 1, level: "info"},
+    {line: 1, level: "error"},
+    {line: 9, level: "warning"},
+    {line: 9, level: "info"},
+    {line: 3, level: "info"},
+    {line: 20, level: "nonsense"}
+  ]
+
+  beforeEach ->
+    sinon.stub(LinterView.prototype, 'initLinters')
+    sinon.stub(LinterView.prototype, 'handleEditorEvents')
+    sinon.stub(LinterView.prototype, 'updateViews')
+    lv = new LinterView(stub_editor)
+
+  afterEach ->
+    LinterView.prototype.initLinters.restore()
+    LinterView.prototype.handleEditorEvents.restore()
+    LinterView.prototype.updateViews.restore()
+
+
+  describe "message sorrting", ->
+
+    it "selects the most severe message for each line", ->
+      lines = lv.sortMessagesByLine(messages)
+      expect(lines['1'].level).to.equal(2)
+      expect(lines['3'].level).to.equal(0)
+      expect(lines['9'].level).to.equal(1)
+
+    it "ignores messages of unrecognized levels", ->
```

Comment:
...not only did you not check for this when I initially thought you missed it, you even have a test! :heart: